src/clib/pioc_support.c

```diff
@@ -2834,7 +2834,7 @@ int PIOc_openfile_retry(int iosysid, int *ncidp, int *iotype, const char *filena
                     ierr = nc_open(filename, file->mode, &file->fh);
                     if(ierr == NC_NOERR)
                     {
-                        printf("PIO: WARNING: Opening file (%s) with iotype=%d failed. Switching iotype to PIO_IOTYPE_NETCDF\n", filename, *iotype);
+                        printf("PIO: WARNING: Opening file (%s) with iotype=%d (%s) failed. But retrying with iotype PIO_IOTYPE_NETCDF was successful. Switching iotype to PIO_IOTYPE_NETCDF for this file\n", filename, *iotype, pio_iotype_to_string(*iotype));
                     }
```

Comment:
Does PIO_IOTYPE_NETCDF get substituted with the actual type?

Reply:
Scorpio internally switches the iotype for the file. However the iotype specified by the user is left unchanged (We tried changing it but it breaks E3SM code).
e.g. The input file is a NETCDF4 file and the user specifies PIO_IOTYPE_PNETCDF as the iotype for opening the file. Since PnetCDF does not support NetCDF4 this call, openfile, will fail and Scorpio will internally retry opening the file with PIO_IOTYPE_NETCDF. If successful the internal (internal to Scorpio) iotype for the file is switched to PIO_IOTYPE_NETCDF, so subsequent reads and writes on the file is performed using the NetCDF library. However the iotype specified by the user is left unchanged (PIO_IOTYPE_NETCDF).

Reply:
> However the iotype specified by the user is left unchanged (PIO_IOTYPE_NETCDF).

Should it be PIO_IOTYPE_PNETCDF here?

Reply:
> > However the iotype specified by the user is left unchanged (PIO_IOTYPE_NETCDF).
> 
> Should it be PIO_IOTYPE_PNETCDF here?

Yes, I just modified my prev comment